targets/actionscript/make.js

```diff
@@ -136,7 +185,13 @@ function getModelPropertyDef(property, datatype) {
     } else {
         if (property.optional && (basicType === "Boolean" || basicType === "int" || basicType === "uint" || basicType === "Number"))
             basicType = "*";
-        return property.name + ":" + basicType;
+        if (property.isclass && fullNamespaceRequired(basicType)) {
+            return property.name + ":" + getValidPlayFabActionScriptNamespacePrefix(datatype) + basicType;
```

Comment:
I'm almost thinking that
getValidPlayFabActionScriptNamespacePrefix
can just return empty string when !fullNamespaceRequired
That way you can simplify this block change.